RelationalAI/Services/Rest.cs

```diff
@@ -265,11 +268,29 @@ private static List<TransactionAsyncFile> ParseMultipartResponse(byte[] content)
             return output;
         }
 
+        private static async Task EnsureSuccessResponseAsync(HttpResponseMessage response)
+        {
+            var status = (int)response.StatusCode;
+            if (status != 404 && status < 500)
+            {
+                return;
+            }
```

Comment:
what happens if resource if forbidden (403) ? Do we swallow client issues other than 404 ?

Reply:
@NRHelmi yeah, we mostly wanted to leave the behaviour as it was before. I think it's better for the SDK team to agree on the http status code handling to be consistent across all of the SDKs.

The reason for 5xx is simple, as we run into them (occasionally we do), we want to be able to see the request id in logs.

Your point is very valid though, but we'd like to leave it out of the scope of this PR is that's fine with you?

Reply:
Thank you for the clarification, can we create an issue as part of the project ?
cc @bradlo